sql-db/sql-app-compatibility/src/test/resources/mariadb_app.properties

```diff
@@ -0,0 +1,7 @@
+quarkus.datasource.db-kind=mariadb
+quarkus.hibernate-orm.dialect=org.hibernate.community.dialect.MariaDBLegacyDialect
+quarkus.datasource.db-version=10.2
```

Comment:
So we set `10.2` while we are using `10.5` for `OpenShiftMariaDBDatabaseIT`, `10.3` for `OpenShiftMariaDB103DatabaseIT`, `DevModeMariadbIT` is using `10.6` and both `MariaDBDatabaseIT` and `DevModeMariadbDevServicesUserExperienceIT` are using`10.10`.

After reading docs, citing:
```
In order for Hibernate ORM to generate more efficient SQL,
to avoid workarounds and to take advantage of more database features,
you can set the database version explicitly
...
As a rule, the version set here should be as high as possible,
but must be lower than or equal to the version of any database your application will connect to.
```

Why not `10.3` then? We could probably take it further and set different version for 2 of these tests so that we test more (that's optional, please just answer 10.3 thingy)

Reply:
I can see the docs says `10.2` is the lowest supported, maybe you want to take it as low as possible?

Reply:
`10.2` has been set here to correspond with the `MariaDB102Dialect`, which was originally used here.
I basically just followed the instructions and replaced `MariaDB102Dialect` with `MariaDBLegacyDialect` + `db-version=10.2`, in order to change the behavior as little as possible and see how the compatibility mode handles it.

Reply:
I think we could do more, but your explanation is fine with me. Thanks